### DISCUSSION TITLE
How to define a custom Markdown syntax?

### DISCUSSION THREAD

**Discussion by iamroylim** [ueberdosis/tiptap#3682] [General Questions] [answered]
example: the "# “ means h1 tag
expect:  See how to define custom Markdown syntax, such as "-" can be converted to a tag

**Comment by sereneinserenade** [comment, marked as answer]
you can use inputRules for that https://tiptap.dev/guide/custom-extensions#input-rules

take a look here https://github.com/ueberdosis/tiptap/blob/main/packages/extension-heading/src/heading.ts#L95 

there is a regex for matching the `#`, you can change it to find `-` or any other char you want to match.

**Reply by iamroylim** [reply]
thank u so much!!!
